### ISSUE TITLE
libinput binaries are non-functional (missing python/evdev?)

### ISSUE THREAD

**Issue by colemickens** [NixOS/nixpkgs#48252]
## Issue description
1. Add `libinput` to your environment packages.

2. `sudo libinput measure touchpad-pressure`:

```
Error: No module named 'evdev'
One or more python modules are missing. Please install those modules and re-run this tool.
```

I'm really not good with python, especially python+nixpkgs. Any suggestions, I can try to fix it?

**Comment by jtojnar** [comment]
I guess we need to use `python3.withPackages (pkgs: with pkgs; [ evdev ])`. It will not work well until https://github.com/NixOS/nixpkgs/pull/46020 is merged though, since meson propagates python which will take precedence.

**Comment by tobiasBora** [comment]
Any news about that issue? Or at least is there any way to temporary fix the issue locally? I tried:

```
[me@nixos:~]$ nix-shell -p libinput -p libevdev -p python3 "python3.withPackages(ps: with ps; [evdev])"

[nix-shell:~]$ sudo libinput measure touchpad-pressure
Error: No module named 'libevdev'
One or more python modules are missing. Please install those modules and re-run this tool.
```

but it does not work...

**Comment by jtojnar** [comment]
I opened https://github.com/NixOS/nixpkgs/pull/92715, however, it bloats libinput maybe too much.

**Comment by tobiasBora** [comment]
Thanks a lot ! Well I think that it's better to have bigger working packages rather than small buggy ones…

I'll test it soon, but out of curiosity, is there some general methods to force temporary an "unpure" fetch of dependencies, in case I'm having similar issues latter on?

Thanks!

**Comment by jtojnar** [comment]
I am not sure what you mean by "unpure" fetch of dependencies. Do you want to replace libinput on your system with the fixed one? In this case the tools do not need to be installed so you should be able to run it using `nix-shell` from this branch. But in general, there is `system.replaceRuntimeDependencies` NixOS option.

**Comment by tobiasBora** [comment]
Interesting option, thanks!

But it's not really what I was asking for, it's more like, imagine that I see that one program/library used by some packages are missing (like here), is there a way to automatically add the program/libraries into the `PATH`/`LD_LIBRARY` variables open in the current `nix-shell`, so that all the program I use automatically have access to that library? For example here I tried to solve the problem with:

```
nix-shell -p libinput -p libevdev -p python3 "python3.withPackages(ps: with ps; [libevdev])"
```

but I guess that the `libinput` program has a wrapper script that removes the `libevdev` library, and makes it unaccessible to `libinput`.

**Comment by jtojnar** [comment]
I would say here your problem is you are defining python twice, and the first one without packages is first in path. Also `nix-shell` does not actually set `LD_LIBRARY_PATH` so `dlopen` will not see it. The following works in this case.

```
nix-shell -I nixpkgs=$PWD -p libinput "python3.withPackages(ps: with ps; [libevdev pyudev])" --run "sudo env LD_LIBRARY_PATH=$(nix-build -A libevdev --no-out-link)/lib libinput measure touchpad-pressure"
```

Of course that is only because https://github.com/NixOS/nixpkgs/commit/de14f0c6e1247aa48b0d77c4a3390a61e020bebf removed Python from `buildInputs`, so the shebang is `/usr/bin/env python3`. If it were processed by `patchShebangsAuto`, you would need to muck with `PYTHONPATH`:

```
nix-shell -I nixpkgs=$PWD -p libinput python3 python3.pkgs.libevdev python3.pkgs.pyudev --run "sudo --preserve-env=PYTHONPATH env LD_LIBRARY_PATH="$(nix-build -A libevdev --no-out-link)/lib" libinput measure touchpad-pressure"
```

And if some wrapper actually sealed the environment variable (usually wrappers just  `--prefix` them but, ocassionally, they use `--set`) you would have no other choice than modify the package – either using `overrideAttrs` or directly editing the expression in your copy of nixpkgs. The latter is what I did when testing the fix, since it is more convenient than trying to add the dependencies as command line arguments.

**Comment by FliegendeWurst** [comment]
This was fixed by #92715.
